src/components/comment/CommentVoteForm.tsx

```diff
@@ -1,72 +1,75 @@
 'use client';
+import React, { useCallback, memo } from 'react';
 import { voteHandlerAction } from '@/actions/commentVote';
 import { useAction } from '@/hooks/useAction';
 import { VoteType } from '@prisma/client';
-import { ThumbsDownIcon, ThumbsUpIcon } from 'lucide-react';
+import ThumbUpOffAltOutlinedIcon from '@mui/icons-material/ThumbUpOffAltOutlined';
+import ThumbDownAltOutlinedIcon from '@mui/icons-material/ThumbDownAltOutlined';
+import ThumbUpIcon from '@mui/icons-material/ThumbUp';
+import ThumbDownIcon from '@mui/icons-material/ThumbDown';
 import { usePathname } from 'next/navigation';
-import React from 'react';
 import { toast } from 'sonner';
 
-const CommentVoteForm = ({
-  upVotes,
-  downVotes,
-  commentId,
-}: {
-  upVotes: number;
-  downVotes: number;
-  commentId: number;
-}) => {
-  const currentPath = usePathname();
-  const { execute } = useAction(voteHandlerAction, {
-    onSuccess: () => {
-      toast('Comment added');
-    },
-    onError: (error) => {
-      toast.error(error);
-    },
-  });
-  const handleUpvote = (e: React.FormEvent<HTMLFormElement>) => {
-    e.preventDefault();
-
-    execute({
-      commentId,
-      voteType: VoteType.UPVOTE,
-      currentPath,
+const CommentVoteForm = memo(
+  ({
+    upVotes,
+    downVotes,
+    commentId,
+  }: {
+    upVotes: number;
+    downVotes: number;
+    commentId: number;
+  }) => {
+    const currentPath = usePathname();
+    const { execute } = useAction(voteHandlerAction, {
+      onSuccess: (voteType) => {
+        console.log(voteType);
+        const message = voteType?.upvotes !== 0 ? 'Upvoted' : 'Downvoted';
+        toast(message);
+      },
+      onError: (error) => {
+        toast.error(error);
+      },
     });
-  };
-  const handleDownVote = (e: React.FormEvent<HTMLFormElement>) => {
-    e.preventDefault();
 
-    execute({
-      commentId,
-      voteType: VoteType.DOWNVOTE,
-      currentPath,
-    });
-  };
-  return (
-    <div className="flex gap-2">
-      <form onSubmit={handleUpvote}>
-        <button
-          className="flex items-center gap-1 text-gray-500 dark:text-gray-400"
-          type="submit"
-        >
-          <ThumbsUpIcon className="w-4 h-4" type="submit" />
+    const UpIcon = upVotes !== 0 ? ThumbUpIcon : ThumbUpOffAltOutlinedIcon;
+    const DownIcon = downVotes !== 0 ? ThumbDownIcon : ThumbDownAltOutlinedIcon;
```

Comment:
I think this is wrong, here upVotes and downvotes are not of the user but a general count. This way you are assuming that the user already voted the filling the icon. It is wrong because if someone upvoted this would result filled for also other users that didnt upvoted

Reply:
Yes, you are correct. While testing the code I only thought about upvotes/downvotes value as 0 or 1. My bad.